src/compiler/diagnostic/index.ts

```diff
@@ -0,0 +1,20 @@
+export interface Diagnostic {
```

Comment:
I created this new sub-project just for the diagnostic file in order to basically make it a 'leaf' node within a future dependency graph - if we actually want to do this more widely throughout the project we'll need to make as many things leaf nodes as possible, otherwise we'll have a really hard time dealing with circular dependency issues (TypeScript does not allow _any_ circular dependencies among project references, unlike within a single project).